paddle/fluid/framework/executor.cc

```diff
@@ -91,7 +91,22 @@ ExecutorPrepareContext::~ExecutorPrepareContext() {
   VLOG(5) << "destroy ExecutorPrepareContext";
 }
 
-Executor::Executor(const platform::Place& place) : place_(place) {}
+Executor::Executor(const platform::Place& place, bool clear_mkldnn_cache)
+    : place_(place), clear_mkldnn_cache_(clear_mkldnn_cache) {}
```

Comment:
@LeoZhao-Intel @jianhang-liu Could you review this PR, it's not elegant to add  `clear_mkldnn_cache` in `Executor` directly, do you have any suggestion?

Reply:
@luotao1  This `clear_mkldnn_cache` was added to prevent clearing cache when we have mulitple predictors e.g. mulitple threads , each thread is having its own predictor (and each predictor's IR Loads Model). If you tell me that this is not needed/used scenario then I can remove `clear_mkldnn_cache` param and clearing cache will always happen in destructor of Executor (this is fine when one predictor , shared by multiple threads). ?

Reply:
> To restore FP32 performance (or even make it a bit better) the only option now is to merge: #20241 . This PR
is having TID in key , but remove many of other attributes that were creating a hashing key. https://github.com/PaddlePaddle/Paddle/issues/20532#issuecomment-542310539

how about remove `clear_mkldnn_cache` param this time to fix FP32 regression ASAP for release/1.6? @jczaja @LeoZhao-Intel @jianhang-liu

Reply:
@jczaja I also don't think adding mkldnn stuff in general "executor" is a good idea. Shall we simply remove this. @LeoZhao-Intel What's your suggestion?

Reply:
I will remove clear_mkldnn_cache param for Executor and that way destructor of Executor will call clearing of mkl-dnn cache regardless any condition. I will update this PR today EOD.

Reply:
@luotao1 I have updated PR. If this PR in acceptable then please test your internal workloads in particular I would like you to test performance of your multi-threaded inference scenario (Multiple threads each doing inference) as I could not test it locally.

Reply:
I test it on https://github.com/PaddlePaddle/Paddle/pull/18372#issuecomment-542992219 multi-thread detect model on the dynamic shape. It seems OK before and after this PR.

Reply:
@luotao1 thank you